test/framework-ondata-mocha.js

```diff
@@ -714,5 +714,223 @@ describe("framework-ondata", function() {
             }).should.eventually.deep.equal(payload);
         });
 
+        it("should propagate input metadata downstream", function() {
+            var testId = "test identifier";
+            var testdata = "more data";
+            return test.createNetwork({
+                node1: {getComponent: factory({inPorts: {input: {}}, 
+                                               outPorts: {output: {}}
+                                             },
+                                             {fRunUpdater: function(vni){ 
+                                                  vni.outputState({id: testId, data: "some data", lm: createLm()}); 
```

Comment:
in this case, yes.  The metadata property can have any legal javascript attribute name.  In these tests, I happen to have used the property name id.  Do you prefer something else?